test/integ/quote-to-ratio.test.ts

```diff
@@ -33,373 +42,829 @@ const callAndExpectFail = async (quoteReq: Partial<QuoteToRatioQueryParams>, res
   }
 }
 
-// Try to parse a user entered amount for a given token
-async function parseAmount(value: number, tokenAddress: string): Promise<string> {
-  const decimals = (await tokenStringToCurrency(tokenAddress))?.decimals
-  return parseUnits(value.toString(), decimals).toString()
+function parseFraction(fraction: ResponseFraction): Fraction {
+  return new Fraction(JSBI.BigInt(fraction.numerator), JSBI.BigInt(fraction.denominator))
 }
 
-async function tokenStringToCurrency(tokenString: string): Promise<Currency | undefined> {
-  const isAddress = (s: string) => s.length == 42 && s.startsWith('0x')
+const SWAP_ROUTER_V2 = '0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45'
 
-  let token: Currency | undefined
-  if (tokenString == 'ETH' || tokenString.toLowerCase() == '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee') {
-    token = Ether.onChain(1)
-  } else if (isAddress(tokenString)) {
-    token = await tokenListProvider.getTokenByAddress(tokenString)
-  }
+describe('quote-to-ratio', async function () {
+  // Help with test flakiness by retrying.
+  this.retries(2)
 
-  if (!token) {
-    token = await tokenListProvider.getTokenBySymbol(tokenString)
-  }
+  this.timeout(10000)
+
+  // chain parameters
+  let alice: SignerWithAddress
+  let block: number
 
-  if (!token) {
-    throw new Error('could not find token')
+  // request parameters
+  let token0: Currency
+  let token1: Currency
+  let token0Balance: CurrencyAmount<Currency>
+  let token1Balance: CurrencyAmount<Currency>
+  let tickUpper: number
+  let tickLower: number
+  let feeAmount: number
+  let slippageTolerance: string
+  let ratioErrorTolerance: number
+  let ratioErrorToleranceFraction: Fraction
+  let response: AxiosResponse<QuoteToRatioResponse>
+
+  function resetQueryParams() {
+    token0 = DAI_MAINNET
+    token1 = USDC_MAINNET
+    token0Balance = parseAmount('5000', DAI_MAINNET)
+    token1Balance = parseAmount('2000', USDC_MAINNET)
+    tickLower = getMinTick(TICK_SPACINGS[FeeAmount.LOW])
+    tickUpper = getMaxTick(TICK_SPACINGS[FeeAmount.LOW])
+    feeAmount = FeeAmount.LOW
+    slippageTolerance = '5'
+    ratioErrorTolerance = 1
+    ratioErrorToleranceFraction = new Fraction(ratioErrorTolerance * 100, 10_000)
   }
```

Comment:
If we do need it, I think it would be cleaner to define them just as 'const quoteToRatioRec: QuoteToRatioQueryParams' so its all typed, or even better just define a 

`const DEFAULT_QUERY_PARAMS = { ... }
`
and make each test override the params that it needs to

`request = { ...DEFAULT_QUERY_PARAMS, token0: ... }`